src/PublishQueuePosixRK.cpp

```
@@ -150,7 +191,43 @@ void PublishQueuePosix::writeQueueToFiles() {
     }
 }
 
+/**
+ * @brief 
+ */
+void PublishQueuePosix::writeEventToFile(const char *eventName, const char *eventData, PublishFlags flags)
+{
+    WITH_LOCK(*this)
+    {
+        PublishQueueEvent *event;
+
+        event = newEvent(eventName, eventData, flags);
```

Are you sure you want to create the newEvent here and not just pass in from publishCommon? The problem here is if newEvent fails (NULL), the error is silenced.

---

Refactored to create the event in PublishCommon and passed event to writeEventToFile and handled return values